packages/playground/src/explorer/views/farms.vue

```diff
@@ -0,0 +1,217 @@
+const totalFarms = ref(0);
+
+const _getFarms = async (queries: Partial<FarmsQuery>) => {
+  loading.value = true;
+  isFormLoading.value = true;
+  try {
+    const { count, data } = await getFarms(queries);
+
+    if (data && count) {
+      totalFarms.value = count;
+      farms.value = data.map(farm => {
+        const ips = farm.publicIps;
+        const total = ips.length;
+        const used = ips.filter(x => x.contract_id === 0).length;
+        return {
+          ...farm,
+          totalPublicIp: total,
+          usedPublicIp: used,
+          freePublicIp: total - used,
+        };
+      });
+      if (mixedFarmFilters.value.options) {
+        if (mixedFarmFilters.value.options.sortBy?.length) {
+          if (sortBy.value[0] && sortBy.value[0]) {
+            const sortKey = sortBy.value[0].key;
+            const sortOrder = sortBy.value[0].order;
+            if (sortKey && sortOrder) {
+              farms.value = farms.value.sort((a, b) => {
+                let aValue: any, bValue: any;
+                if (sortKey == "farmId") {
+                  aValue = a.farmId;
+                  bValue = b.farmId;
+                } else {
+                  aValue = a.name;
+                  bValue = b.name;
+                }
+
+                if (typeof aValue == "string" && typeof bValue == "string") {
+                  return sortOrder === "desc" ? bValue.localeCompare(aValue) : aValue.localeCompare(bValue);
+                }
+                return sortOrder === "desc" ? bValue - aValue : aValue - bValue;
+              });
+            }
+          }
+        }
+      }
+    }
+  } catch (err) {
+    console.log("could not get farms:", err);
+    createCustomToast("Failed to get farms!", ToastType.danger);
+  } finally {
+    isFormLoading.value = false;
+    loading.value = false;
+  }
+};
+onMounted(async () => {
+  await _getFarms({});
+
+  await updateFarms();
+});
+
+const request = debounce(_getFarms, 1000);
+const updateFarms = async () => {
+  const queries = getFarmQueries(mixedFarmFilters.value);
+
+  await request(queries);
+};
+
+const updateQueries = () => {
+  const options = mixedFarmFilters.value.options;
+  if (options) {
+    options.page = page.value;
+    options.size = size.value;
+    options.sortBy = sortBy.value;
+  }
+};
+watch(mixedFarmFilters.value, updateFarms, { deep: true });
+
+const inputFiltersReset = (nFltrNptsVal: FilterFarmInputs) => {
+  mixedFarmFilters.value.inputs = nFltrNptsVal;
+  nFltrNptsVal.farmId.value = undefined;
+  nFltrNptsVal.name.value = undefined;
+  nFltrNptsVal.pricingPolicyId.value = undefined;
+};
+const openSheet = (_e: any, { item }: any) => {
+  openDialog(item.value);
+};
+const openDialog = (item: Farm) => {
+  selectedFarm.value = item;
+  isDialogOpened.value = true;
+};
+
+const closeDialog = () => {
+  isDialogOpened.value = false;
+};
+const headers: VDataTableHeader = [
+  { title: "ID", key: "farmId" },
+  { title: "Name", key: "name" },
```

Comment:
Sorting is working now but there's a weird behavior, if i switched from the first page and sorted the id or name it will then return to the first page and sort the values that are available there instead of sorting the values in the selected page.

[Screencast from 21-11-23 15:23:40.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/9f1a1e26-8e2f-464b-b2a1-b88d4810e90f)

Reply:
it is actually not weird behaviour to go back to page 1 when sorting is selected please refer to vuetify server side table [docs](https://vuetifyjs.com/en/components/data-tables/server-side-tables/#server-side-paginate-and-sort)

Reply:
Indeed after checking the [docs](https://vuetifyjs.com/en/components/data-tables/server-side-tables/#data-table-server-side-tables) i found that this is the normal behavior for server side tables in vuetify